web-animations/timing-model/animations/updating-the-finished-state.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>Tests for updating the finished state of an animation</title>
+<link rel="help" href="https://w3c.github.io/web-animations/#updating-the-finished-state">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../../testcommon.js"></script>
+<body>
+<script>
+'use strict';
+
+function validateFinishEvent(t, event, animation) {
+  t.step(function() {
+    assert_equals(event.target, animation, "Animation should be event target");
+    assert_times_equal(event.currentTime, animation.currentTime, "Event currentTime should be animation currentTime");
+    assert_times_equal(event.timelineTime, animation.timeline.currentTime, "Event timelineTime should be animation timeline's currentTime");
+  });
+}
+
+function awaitFinishEventAndPromise(t, animation) {
+  var eventPromise = new Promise(function(resolve) {
+    animation.onfinish = function(event) {
+      validateFinishEvent(t, event, animation);
+      resolve();
+    };
+  });
+
+  var finishedPromise = animation.finished.then(function(target) {
+    assert_equals(target, animation);
+  });
+
+  return Promise.all([eventPromise, finishedPromise]);
+}
+
+async_test(function(t) {
+  var div = createDiv(t);
+  var animation = div.animate(null, 1);
+  animation.onfinish = function(event) {
+    t.unreached_func("Seeking to finish should not fire finish event");
```

Comment:
This doesn't actually do anything. It just creates a function, but never calls it. Hence this will never fail.